sklearn/pipeline.py

```diff
@@ -609,12 +629,26 @@ def _transform_one(transformer, X, y, weight, **fit_params):
     return res * weight
 
 
-def _fit_transform_one(transformer, X, y, weight, **fit_params):
-    if hasattr(transformer, 'fit_transform'):
-        res = transformer.fit_transform(X, y, **fit_params)
-    else:
-        res = transformer.fit(X, y, **fit_params).transform(X)
-    # if we have a weight for this transformer, multiply output
+def _fit_transform_one(is_transform, clsname, message, transformer, weight, X,
+                       y, **fit_params):
+    """
+    Fits ``transformer`` to ``X`` and ``y``.
+
+    If ``is_transform`` is ``True``, then ``X``, ``y`` will be transformed. The
+    transformed result is returned with the fitted transformer. If ``weight``
+    is not ``None``, the result will be multipled by ``weight``.
+
+    If ``is_transform`` is ``False``, then a tuple of
+    (``None``, fitted_transformer) will be returned.
+    """
+    with log_elapsed(clsname, message):
+        if not is_transform:
+            return None, transformer.fit(X, y, **fit_params)
```

Comment:
not so keen on this one. Why is this `if` branch needed for logging: seems like it could fit under the below logic?

Reply:
There are log messages for the `is_transform==True` and `is_transform==False`. In this case, when `is_transform==False` only the fitted transformer is returned. The fitting procedure is logged in this case.

I could be missing your point. Where do you think would be a better place to put the logging context manager?